public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -58,25 +59,27 @@ interface State {
 }
 
 export class DataSourceSettingsPage extends PureComponent<Props, State> {
+  importDataSourcePlugin: typeof importDataSourcePlugin;
+  importPluginPromise: CancelablePromise<any> = null;
+
   constructor(props: Props) {
     super(props);
 
     this.state = {
       plugin: props.plugin,
     };
+
+    this.importDataSourcePlugin = importDataSourcePlugin;
```

Comment:
Why attaching this imported function to an instance? I don't see a need for this

Reply:
I did this because I I make the instance version cancelable. I was thinking that if the props are updated, is the cancelable version not replaced? It was also done here in `PromQueryField`: https://github.com/grafana/grafana/blob/45b7de1910819ad0faa7a8aeac2481e675870ad9/public/app/plugins/datasource/prometheus/components/PromQueryField.tsx#L130